source/standards/storing-credentials.html.md.erb

```diff
@@ -0,0 +1,48 @@
+
+# <%= current_page.data.title %>
+
+Depending on how you [manage your accounts](/standards/accounts-with-third-parties.html), you, your team and the service you run may have credentials or other secrets that you need to store securely.
+
+## Personal credentials
+
+Examples include login details for:
+
+- GitHub
+- AWS
+- GOV.UK Signon
+
+[Use the password manager built into to your browser in the first instance](https://lock.cmpxchg8b.com/passmgrs.html#conclusion).
+
+If you can't do this for some reason, then it's OK to use a third-party password manager. This could be necessary if your browser has an accessibility issue, or if you work with multiple browsers.
+
+Some of the third-party password managers people use at GDS include:
+
+- **LastPass** - Approved by Information Assurance and available to install in the GDS Self Service app. However, you can only use the free version on one type of device.
+- [**1Password**](https://1password.com/)
+- [**BitWarden**](https://bitwarden.com/) - An open source password manager.
+- [**KeePassXC**](https://keepassxc.org/) -  An offline password store, which you may want to backup somewhere.
+
+**Do not use browser extensions that autofill credentials**, as [these are fundamentally vulnerable to attack](https://lock.cmpxchg8b.com/passmgrs.html#interprocess-communication). If you do use an extension, make sure it's configured so that you manually copy / paste credentials.
+
+## Team credentials
+
+Examples include shared login details for:
+
+- Software repositories e.g. npmjs.org, rubygems.org
+- Admin portals e.g. Fastly, DockerHub
+
+There's no established best practice for storing shared credentials at GDS. Some teams have a "credentials" GitHub repository and use [pass](https://www.passwordstore.org/) to store credentials encrypted with GPG. However, [this is fundamentally insecure](https://latacora.micro.blog/2019/07/16/the-pgp-problem.html) and we strongly encourage you to look around for possible alternatives. **Any GitHub repositories that contain secrets should be [internal or ideally private](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/creating-a-repository-on-github/about-repository-visibility)** to help limit access by former staff.
+
+## Service credentials
+
+Examples include:
+
+- Functional account credentials e.g. API keys
+- Sensitive configuration (e.g. IP block lists)
+
+Some teams have "secrets" repositories and use GPG to encrypt files, which are then decrypted during the deployment process. As above, [storing secrets this way is fundamentally insecure](https://latacora.micro.blog/2019/07/16/the-pgp-problem.html) and we strongly encourage you to look around for possible alternatives. **Any GitHub repositories that contain secrets should be [internal or ideally private](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/creating-a-repository-on-github/about-repository-visibility)** to help limit access by former staff.
```

Comment:
I don't think there's much value in duplicating the content above:

```suggestion
Some teams have "secrets" repositories and use GPG to encrypt files, which are then decrypted during the deployment process. As above, this is not very secure and we encourage you to look around for possible alternatives.
```

Reply:
I would suggest best practice here would be to use the secrets management infrastructure provided by your infrastructure or cloud provider. e.g. AWS Secrets Manager, or HashiCorp Vault.

Again, I think "fundamentally insecure" is too strong for GPG. It's more "fundamentally a pain in the arse".

Reply:
Both reasonable comments, thanks. I duplicated the warnings because they may evolve separately, which is happening already. I didn't want to end up with "as above" (when there is no longer an "above").

How about this?

> Use the secrets management feature of your infrastructure or cloud provider e.g. AWS Secrets Manager, HashiCorp Vault. This should make it easy to control and audit access to the credentials.
>
> Older (pre-cloud) projects in GDS tend to have a "secrets" repository and use GPG to encrypt files, which are then decrypted during the deployment process. However, using GPG [isn't ideal](https://latacora.micro.blog/2019/07/16/the-pgp-problem.html).

It would be nice if we had some links to further guidance or examples of how we use cloud secrets management. I'm aware of [this blog](https://technology.blog.gov.uk/2021/07/23/automatically-creating-and-rotating-gov-uk-application-secrets/), but maybe the provider docs are a better starting point. What do you think?

Reply:
Yeah, I'm not sure if there's a great place to link. That blog is nice, but it's getting into automating rotations and maybe that's a bit more advanced than most people reading this guidance need. I guess if you can find some AWS / Hashi docs which are high enough level we could link to those.